vscode-extension/PSA_custom_Rules/Alias/avoidAlias.psm1

```diff
@@ -0,0 +1,92 @@
+#Requires -Version 3.0
+
+<#
+.SYNOPSI
+    Avoid alias in powershell script.
+.DESCRIPTION
+    Find all aliases that appear in the powershell script. And give the suggestion to change them to formal name.
+.EXAMPLE
+    Measure-AvoidAlias -ScriptBlockAst $ScriptBlockAst
+.INPUTS
+    [System.Management.Automation.Language.ScriptBlockAst]
+.OUTPUTS
+    [OutputType([Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord[]])]
+.NOTES
+    None
+#>
+function Measure-AvoidAlias {
+    [CmdletBinding()]
+    [OutputType([Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord[]])]
+    Param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.Management.Automation.Language.ScriptBlockAst]
+        $testAst
+    )
+    
+    Process {
+        $results = @()
+        # import functions
+        $findCmdFunctionFile = ".\PSA_custom_Rules\Find-CmdletsInFile.psm1"
+        Import-Module $findCmdFunctionFile
+        $getAliasSpecFunctionFile = ".\PSA_custom_Rules\Alias\Get-AliasSpec.psm1"
+        Import-Module $getAliasSpecFunctionFile
+
+        #get the alias mapping data
+        $aliasSpecFile = ".\PSA_custom_Rules\Alias\AliasSpec.json"
+        $AliasSpec = Get-AliasSpec -AliasPath $aliasSpecFile
+
+        # get the commandAst in the file
+        $foundCmdlets = Find-CmdletsInFile -rootAstNode $testAst
+    
+        #list of CorrectionExtents
+        $l = (new-object System.Collections.ObjectModel.Collection["Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.CorrectionExtent"])
+
+        foreach ($cmdletReference in $foundCmdlets) {
+            if ($AliasSpec.cmdlet.psobject.properties.match($cmdletReference.CommandName).Count) {
```

Comment:
Please use proper casing
```suggestion
            if ($AliasSpec.Cmdlet.PSObject.Properties.Match($cmdletReference.CommandName).Count) {
```

Reply:
"cmdlet" is custom key in spec renamed as "cmdlet", so I think it should be "cmdlet" not "Cmdlet" here.

Reply:
OK.